### ISSUE TITLE
Consider enabling overlinking checks for conda builds in RAPIDS

### ISSUE THREAD

**Issue by vyasr** [rapidsai/build-planning#21]
Enabling overlinking checks will help us capture issues with things like rpaths or libraries that we're implicitly/unexpectedly linking to.

**Comment by vyasr** [comment]
Based on my reading of https://rattler.build/latest/build_options/#dynamic-linking-configuration, rattler-build has overlinking checks on by default. Therefore, I think that this will be automatically resolved once #47 is complete. Please correct me if I'm misunderstanding though. @gforsyth it would help if for packages other than rmm (i.e. libraries that are no header-only) we could double-check the rattler logs and see that we are indeed passing overlinking/overdepending checks.

**Comment by gforsyth** [comment]
Yep, I can confirm that the overlinking and overdepending checks are on by default.  There was some chatter on the prefix.dev discord channel about them not being quite "perfect" but that was also 8 or 9 months ago.

**Comment by vyasr** [comment]
Awesome. I'm going to close this as covered by #47, then.
